tests/e2escenarios/e2e_devfile_test.go

```diff
@@ -0,0 +1,174 @@
+package e2escenarios
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/odo/tests/helper"
+
+	. "github.com/onsi/ginkgo"
+)
+
+// Test Objective:
+//    Test ODO devfile support features
+
+// Scope:
+//    Test debug support for the following components, making use of starter projects define in the corresponding devfile:
+//    - nodejs
+//    - java-springboot
+//    - java-quarkus
+//    - java-openliberty
+//    - java-maven, no starter project available at the time this script was developed, so skiping this component for now
+
+var _ = Describe("odo devfile supported tests", func() {
+	var componentName, projectDirPath string
+	var projectDir = "/projectDir"
+	var commonVar helper.CommonVar
+
+	// This is run before every Spec (It)
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		componentName = helper.RandString(6)
+		helper.Chdir(commonVar.Context)
+		projectDirPath = commonVar.Context + projectDir
+	})
+
+	preSetup := func() {
+		helper.MakeDir(projectDirPath)
+		helper.Chdir(projectDirPath)
+	}
+
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	Context("odo debug on a nodeJS starterProject component", func() {
+		JustBeforeEach(func() {
+			preSetup()
+		})
+
+		It("Verify output debug information for nodeJS debug works", func() {
+			fmt.Println("********************************************************")
+			fmt.Println("*   Test devfile debug support for nodeJS component    *")
+			fmt.Println("********************************************************")
+			helper.CmdShouldPass("odo", "create", "nodejs", "--starter", "--project", commonVar.Project, componentName, "--context", projectDirPath)
```

Comment:
It seems we are running the similar steps for each and every context. I would suggest to create a common function with all the steps like https://github.com/openshift/odo/blob/master/tests/e2escenarios/e2e_images_test.go#L36 and within a single `Context` run all the `It` spec with the created function.

For reference you can use https://github.com/openshift/odo/blob/master/tests/e2escenarios/e2e_images_test.go test script.

Reply:
Addressed in commit "#3377 addressed all Priti's comments in PR"

Reply:
comment was removed from code